### PR TITLE
Certain pixeldepth-16 TGA files fail to process, classcast exception

### PR DESCRIPTION
**Pull request by KoenDG**
I added the 2 files for which I encountered this.

Apparently `getDataBuffer` can also return `DataBufferByte` for images with pixeldepth of 16.

Not sure if the comments I left in the test code are correct or clear enough.

Running the tests without the fixed code produces:

```
testRead(com.twelvemonkeys.imageio.plugins.tga.TGAImageReaderTest)  Time elapsed: 0.059 sec  <<< FAILURE!
java.lang.AssertionError: Image file:<REDACTED>/TwelveMonkeys/imageio/imageio-tga/target/test-classes/tga/monochrome16_top_left.tga index 0 could not be read: java.lang.ClassCastException: java.awt.image.DataBufferByte cannot be cast to java.awt.image.DataBufferUShort
	at com.twelvemonkeys.imageio.util.ImageReaderAbstractTest.failBecause(ImageReaderAbstractTest.java:105)
	at com.twelvemonkeys.imageio.util.ImageReaderAbstractTest.testRead(ImageReaderAbstractTest.java:224)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43
```

### CODE REVIEW

File: imageio/imageio-tga/src/main/java/com/twelvemonkeys/imageio/plugins/tga/TGAImageReader.java [hunk 1]
```diff
@@ -196,8 +196,15 @@ public BufferedImage read(final int imageIndex, final ImageReadParam param) thro
...
-                    readRowUShort(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataUShort, destRaster, clippedRow, y);
+                    final DataBuffer dataBuffer = rowRaster.getDataBuffer();
+                    if (dataBuffer instanceof DataBufferByte) {
+                        byte[] rowDataByte16 = ((DataBufferByte) rowRaster.getDataBuffer()).getData();
+                        readRowByte(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataByte16, destRaster, clippedRow, y);
+                    } else {
+                        // DataBufferUShort and any other
+                        short[] rowDataUShort = ((DataBufferUShort) rowRaster.getDataBuffer()).getData();
+                        readRowUShort(input, height, srcRegion, header.getOrigin(), xSub, ySub, rowDataUShort, destRaster, clippedRow, y);
+                    }
```
Comment (haraldk) [inline imageio/imageio-tga/src/main/java/com/twelvemonkeys/imageio/plugins/tga/TGAImageReader.java:207]: Good catch! There is certainly a bug in the current `TGAImageReader` concerning 16 bit unsigned gray.

The suggested fix, however, doesn't properly read the 16 bit values. 

Instead, we can change the `getRawImageType`, to handle the fact that there is such a thing as 16 bit gray samples.

Line 119 and on:

```java
            case TGA.IMAGETYPE_MONOCHROME:
            case TGA.IMAGETYPE_MONOCHROME_RLE:
                switch (header.getPixelDepth()) {
                    case 8:
                        return ImageTypeSpecifiers.createFromBufferedImageType(BufferedImage.TYPE_BYTE_GRAY);
                    case 16:
                        return ImageTypeSpecifiers.createFromBufferedImageType(BufferedImage.TYPE_USHORT_GRAY);
                    default:
                        throw new IIOException("Unknown pixel depth for monochrome: " + header.getPixelDepth());
                }
```

**Review by haraldk — CHANGES_REQUESTED** [review]
Based on your find and sample files, I also found a bug in the metadata, that reports 5,5,5 for bits per sample... I can fix that and the `getRawImageType`method, if you roll back the changes in the `read` method. 😀

**Comment by KoenDG** [comment]
Code updated as requested.

**Review by haraldk — APPROVED** [review]
👍🏻

**Pull request merged**